src/v/cloud_topics/app.cc

```diff
@@ -85,10 +85,9 @@ ss::future<> app::construct(
       manager,
       ss::sharded_parameter([&] { return &remote->local(); }),
       bucket,
-      ss::sharded_parameter(
-        [&] { return &controller->get_partition_manager().local(); }),
-      ss::sharded_parameter(
-        [&] { return &controller->get_raft_manager().local(); }));
+      &controller->get_partition_manager(),
+      &controller->get_raft_manager(),
+      &controller->get_health_monitor());
```

Comment:
bitten by start-up order issues. `get_health_monitor.local()` isn't valid until `app:start` is called (based on the way that `application.cc` orders operations between the CT manager and the controller).

not as "clean", but switch to passing around the entire sharded service since we can't pluck out the local core reference at this point.

Reply:
I wish there was a better solution for this :/